libraries/Microsoft.Bot.StreamingExtensions/ContentStream.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using Microsoft.Bot.StreamingExtensions.Payloads;
+
+namespace Microsoft.Bot.StreamingExtensions
+{
+    internal class ContentStream : IContentStream
+    {
+        private readonly ContentStreamAssembler _assembler;
+        private PayloadStream _stream;
+
+        internal ContentStream(Guid id, ContentStreamAssembler assembler)
+        {
+            Id = id;
+            _assembler = assembler ?? throw new ArgumentNullException();
+        }
+
+        public Guid Id { get; private set; }
+
+        public string Type { get; set; }
+
```

Comment:
"Type" is a C# Keyword. We should really avoid using similar names.

Reply:
That's very true. I'll give this a lot of thought. The problem with any changes to ContentStream is it will either break consistency between it and StreamDescription.cs or, if both are changed, cause a complete breaking change for all users with an older version of the code. I'm checking in to how large of a blast radius that would be.

Reply:
After digging I've found we alias the name on both the incoming and outgoing paths, so I've renamed it ContentType, because it's storing the value of the header ContentType.